subprojects/docs/src/snippets/buildCache/integration-tests/groovy/build.gradle

```diff
@@ -27,38 +27,42 @@ tasks.named('integTest') {
 // end::distributionPathInput[]
 
 // tag::distributionDirInput[]
-class DistributionLocationProvider implements CommandLineArgumentProvider { // <1>
+abstract class DistributionLocationProvider implements CommandLineArgumentProvider {  // <1>
     @InputDirectory
-    @PathSensitive(PathSensitivity.RELATIVE)                                // <2>
-    File distribution
+    @PathSensitive(PathSensitivity.RELATIVE)  // <2>
+    abstract DirectoryProperty getDistribution()
 
     @Override
     Iterable<String> asArguments() {
-        ["-Ddistribution.location=${distribution.absolutePath}".toString()] // <3>
+        ["-Ddistribution.location=${distribution.get().asFile.absolutePath}".toString()]  // <3>
```

Comment:
❓ is the `.toString()` strictly necessary here?

Reply:
> is the `.toString()` strictly necessary here?

Initial author was probably overly cautious with GStrings, but removing `toString` seems to work well.